File: backend/services/llm.py
from openai import AsyncOpenAI

from backend.config import settings


class LLMClient(AsyncOpenAI):
    """Thin wrapper around AsyncOpenAI to store the model name."""

    def __init__(self):
        super().__init__(
            base_url=settings.llm_base_url,
            api_key=settings.llm_api_key,
        )
        self._model = settings.llm_model


llm_client = LLMClient()

File: backend/services/llm.ts
import OpenAI from "openai";
import { settings } from "../config.js";

class LLMClient extends OpenAI {
  _model: string;

  constructor() {
    super({
      baseURL: settings.llmBaseUrl,
      apiKey: settings.llmApiKey,
    });

    this._model = settings.llmModel;
  }
}

export const llmClient = new LLMClient();
